src/pat/validation/validation.js

```diff
@@ -58,20 +58,21 @@ define([
             this.errors = 0;
             this.options = parser.parse(this.$el, opts);
             this.$inputs = this.$el.find(':input[name]');
-            this.$el.find(":input[type=number]").on('keyup mouseup', _.debounce(function (ev) {
+
+            // Fire - debounced - when value of input elements changes
+            // See: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event
+            this.$el.find("input:not([type=button]):not([type=reset]):not([type=submit]:not([type=checkbox]):not([type=radio])), select, textarea, *[contenteditable]:not([contenteditable=false])").on('input', _.debounce(function (ev) {
+                this.validateElement(ev.target);
+            }.bind(this), 500));
+            // See: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event
+            this.$el.find("input[type=checkbox]), input[type=radio]").on('change', _.debounce(function (ev) {
                 this.validateElement(ev.target);
             }.bind(this), 500));
-            this.$inputs.on('change.pat-validation', function (ev) { this.validateElement(ev.target); }.bind(this));
-            // formaction causes form validation to be skipped, so validate on
-            // submit button click instead.
-            this.$el.find('button[type=submit][formaction]').on('click.pat-validation', this.validateForm.bind(this));
-            this.$el.on('submit.pat-validation', this.validateForm.bind(this));
+
+            // Validate on form submit.
+            this.$el.on('submit', this.validateForm.bind(this));
```

Comment:
Just validate on form submit.
If invalid, cancel current submit.
Specifically handle AJAX submission (would be nice, if they also hook into standard submit events).
Do not validate on form close.